### ISSUE TITLE
delete() on changefeeds

### ISSUE THREAD

**Issue by tatsujin1** [rethinkdb/rethinkdb#5772]
Wouldn't it be neat if it was possible to "hang" `delete()` on a changefeed, meaning that all documents emitted by the changefeed are deleted.

* #746 in combo with #3583 would then becomes simple, explicit and easy to understand:
`table.filter(...).changes().delete()`
* It becomes very similar to the non-changefeed delete:
`table.filter(...).delete()`

I guess this can be seen as a syntax suggestion to the implementation of #746...

**Comment by danielmewes** [comment]
Thanks for the suggestion @tatsujin1 .

We've had some similar discussions before, but as far as I can find we hadn't had an issue for this.

One issue with this kind of thing (having a write operation chained onto a changefeed) is that you can easily build infinite loops, where the write triggers another change and thereby gets triggered over and over.

For the specific case of TTL we'll probably provide a different framework (though the API you describe would be very nice for it indeed!). I've been wanting to write up a proposal for it for the past week, but have been busy with Horizon-related things. Will try to get to it today or tomorrow.

**Comment by mlucy** [comment]
@tatsujin1 -- you can write `.changes().foreach{|change| table.write_operation_based_on_change}` to get something like that.

**Comment by danielmewes** [comment]
@mlucy I believe I had tried that at some point and it was forbidden.

**Comment by mlucy** [comment]
You're right; it looks like we forbid calling `forEach` on an infinite stream.

**Comment by tatsujin1** [comment]
I guess it makes sense - to be consistent - to allow all write operations. But as you mentioned infinite loops tend to be bad :) I guess it could even disrupt things elsewhere, like hogging the server.

On the other hand, most programming languages allow writing infinite loops, and as a programmer that foot is shot from time to time ;)
